src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +69,68 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
+    const ValueType initial(result);
     Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    result += initial;
   }
 }
 
+/*
+ * Computes a reduction over the view described by 'input' for entries 'input([begin,end))'
+ * The variable 'input' should be an operator s.t. 'input(k)' returns a Pack of size 'vector_size'.
+ * The variable 'total_size' represents the total number of 'ValueType' entries in the view. The
+ * computed reduction is added to 'result'.
+ */
+template <bool Serialize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int total_size,
+                     const int vector_size,
```

Comment:
I think we can omit `vector_size` by doing something like `static consexpr int vector_size = typename decltype(InputProvider)::n` in the impl.

Reply:
Input provider is not a pack, but `input(0)` is, so `decltype(input(0))` will do.

Reply:
I commented on this below.